src/main/java/marquez/db/DatasetDao.java

```diff
@@ -0,0 +1,66 @@
+package marquez.db;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import marquez.common.models.DatasetUrn;
+import marquez.common.models.Namespace;
+import marquez.db.mappers.DatasetRowMapper;
+import marquez.db.models.DataSourceRow;
+import marquez.db.models.DatasetRow;
+import marquez.db.models.DbTableInfoRow;
+import marquez.db.models.DbTableVersionRow;
+import marquez.db.sql.BindDatasetRow;
+import org.jdbi.v3.sqlobject.CreateSqlObject;
+import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.customizer.BindBean;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+import org.jdbi.v3.sqlobject.transaction.Transaction;
+
+@RegisterRowMapper(DatasetRowMapper.class)
+public interface DatasetDao {
+  @CreateSqlObject
+  DataSourceDao createDataSourceDao();
+
+  @CreateSqlObject
+  DbTableVersionDao createDbTableVersionDao();
+
+  @SqlUpdate(
+      "INSERT INTO datasets (uuid, namespace_uuid, datasource_uuid, urn, description) "
+          + "VALUES (:uuid, :namespace_uuid, :datasource_uuid, :urn, :description)")
+  void insert(@BindDatasetRow DatasetRow datasetRow);
```

Comment:
what does `@BindDatasetRow` offer that `@BindBean` wouldnt? If we carry this pattern forward, will we need to write custom `Bind*Row` and `Bind*RowFactory` classes for every DAO model? Worth the calories?